benchmarks/benchmarks/analysis/distances.py

```diff
@@ -11,120 +17,243 @@
-    # unit cell information in the benchmarks
-    params = (10, 100, 1000, 10000)
-    param_names = ['num_atoms']
+    params = ([10, 100, 1000, 10000])
+    param_names = (['num_atoms'])
 
     def setup(self, num_atoms):
+
+        self.u = MDAnalysis.Universe(GRO)
+
+        self.ag1 = self.u.atoms[:num_atoms]
+        self.ag2 = self.u.atoms[num_atoms: 2 * num_atoms]
+        self.ag3 = self.u.atoms[-num_atoms:]
+
         np.random.seed(17809)
         self.coords_1 = np.random.random_sample((num_atoms, 3)).astype(np.float32)
         np.random.seed(9008716)
         self.coords_2 = np.random.random_sample((num_atoms, 3)).astype(np.float32)
         self.allocated_array_2D = np.empty((num_atoms, num_atoms),
-                                            dtype=np.float64)
+                                           dtype=np.float64)
         self.array_shape_1D = int(num_atoms * (num_atoms - 1) / 2.)
         self.allocated_array_1D = np.empty(self.array_shape_1D,
                                            dtype=np.float64)
+
+    def time_between(self, num_atoms):
+        """Benchmark determination of subgroup
+        of atomgroup that is within a specific
+        distance of two other atomgroups.
+        """
+        distances.between(group=self.ag3,
+                          A=self.ag1,
+                          B=self.ag2,
+                          distance=15.0)
+
+
+class DistancesBench(object):
+    """Benchmarks for MDAnalysis.analysis.distances
+    functions. Excluding contact matrices.
+    """
+    u = MDAnalysis.Universe(GRO)
+    tri_clinic_dimensions = u.dimensions
+
+    params = ([10, 100, 1000, 10000], [None, 'orthogonal', 'triclinic'])
+    param_names = ['num_atoms', 'pbc_type']
+
+    def setup(self, num_atoms, pbc_type):
+
         self.u = MDAnalysis.Universe(GRO)
+
+        if pbc_type is None:
+            self.box_dims = None
+
+        elif pbc_type == 'orthogonal':
+            box_shape = self.u.dimensions
+            basis_vectors = mdamath.triclinic_vectors(box_shape)
+
+            # this will calculate the furthest point from the origin on
+            # the triclinic unit cell.
+            furthest_point = np.sum(basis_vectors, axis=0)
+            # get full size of cube and also deal with certain distance
+            # functions requiring float32
+            orthogonal_box_size = np.float32(np.max(furthest_point))
+            angle = np.float32(90.0)
+
+            # make dummy cubic box, must be np.array, does not accept list
+            self.box_dims = np.array([orthogonal_box_size,
+                                      orthogonal_box_size,
+                                      orthogonal_box_size,
+                                      angle, angle, angle])
+
+        elif pbc_type == 'triclinic':
+            self.box_dims = self.u.dimensions
+
+        else:
+            raise Exception("Specify only 'orthogonal', 'triclinic', or None for the box type")
```

Comment:
Here and elsewhere it seems a bit unusual to raise a generic exception (or any exception, really) in the `setup()` of a benchmark. Since the `asv` benchmarks are not "public" and we control the parameters, plus if the box type is invalid, we should already have error handling for it. `asv` will tend to handle exceptions fairly gracefully, reporting results for benchmarks that do not fail, and filling the tables in with failure entries for those benchmarks that fail.

Reply:
Yes, I thought it might be odd. I'll get rid of them.